tests/testthat/test-opencl.R

```diff
@@ -0,0 +1,109 @@
+if (not_on_cran()) {
+  set_cmdstan_path()
+  fit <- testing_fit("bernoulli", method = "sample", seed = 123, chains = 1)
+}
+
+test_that("all methods error when opencl_device is used with non OpenCL model", {
+  skip_on_cran()
+  stan_file <- testing_stan_file("bernoulli")
+  mod <- cmdstan_model(stan_file = stan_file)
+  expect_error(
+    mod$sample(data = testing_data("bernoulli"), opencl_device = c(0, 0), chains = 1),
+    "'opencl_device' is set but the model was not compiled with for use with OpenCL.",
+    fixed = TRUE
+  )
+  expect_error(
+    mod$optimize(data = testing_data("bernoulli"), opencl_device = c(0, 0)),
+    "'opencl_device' is set but the model was not compiled with for use with OpenCL.",
+    fixed = TRUE
+  )
+  expect_error(
+    mod$variational(data = testing_data("bernoulli"), opencl_device = c(0, 0)),
+    "'opencl_device' is set but the model was not compiled with for use with OpenCL.",
+    fixed = TRUE
+  )
+  stan_file_gq <- testing_stan_file("bernoulli_ppc")
+  mod_gq <- cmdstan_model(stan_file = stan_file_gq)
+  expect_error(
+    mod_gq$generate_quantities(fitted_params = fit, data = testing_data("bernoulli"), opencl_device = c(0, 0)),
+    "'opencl_device' is set but the model was not compiled with for use with OpenCL.",
+    fixed = TRUE
+  )
+})
+
+test_that("all methods error on invalid opencl_device", {
+  skip_on_cran()
+  skip_if(Sys.getenv("CMDSTANR_OPENCL_TESTS")!="1")
```

Comment:
If CMDSTANR_OPENCL_TESTS is set in the developers systems these will run. Is this OK?

Reply:
This seems okay to me. Presumably this will wait for Jonah. Could the fixed c(0, 0) deviceIDs be a problem?

Reply:
This wont run on CI and is mostly just a sanity check. We can wait for @jgabry but idt is such a big thing.

Reply:
Yeah this seems fine